server/routes/songs.js

```diff
@@ -256,13 +269,32 @@ router.post("/globallysearched/add", function (req, res, next) {
   return res.status(201).json(data);
 });
 
-// todo
 // PUT update like (for scoreboard)
-router.put("/likes/update", function (req, res, next) {
-  const songID = req.body.songID;
-  const isLiked = req.body.isLiked;
-  // update database
-  return res.status(201).json({ songID: songID, like: isLiked });
+router.put("/likes/update", async function (req, res) {
+  const { song, isLiked } = req.body;
+
+  try {
+    let likedSong = await LikedSongs.findOne({ spotifyId: song.id });
+    if (likedSong) {
+      if (isLiked) {
+        likedSong.likes++;
```

Comment:
should we be also subtracting from the number if they decide to unlike it, after liking it?

Reply:
I think this would only make sense if we have user profiles setup for everyone and we keep track of the songs that the user has liked. Currently, the way I understand it is that the likes (red hearts) are only stored while the user is on the page. So when the page is refreshed everything disappears. For the purpose of a global scoreboard, I think this should be fine.

Reply:
@rrichc @JeromeTing @KDhieb any thoughts? i feel like intuitively i should be able to reverse my like in that same session if i wanted to, could argue either way though (with our current setup)

Reply:
Yeah, it's up to you guys. It should be pretty easy to implement

Reply:
I agree both are valid. If the page is refreshed it makes sense that the likes may be lost and there's no way to "unheart" a song, but in a session it seems intuitive to unheart a song if they made a mistake or something like that. 

Alternatively, we could do a thumbs up vs thumbs down approach instead since we might not implement users?

Reply:
We could definitely add the subtraction to prevent abuse of the scoreboard and to avoid inflated likes (although not super important). Also from the user perspective, hearting/unhearting a song doesn't provide much value aside from a UI change (and increasing the scoreboard counter).

Reply:
I've set it up so that when the user unhearts a song a like is subtracted from that song